app/javascript/responseTemplates/responseTemplates.js

```diff
@@ -179,9 +176,34 @@ function fetchResponseTemplates(typeOf, formId) {
     .then((response) => response.json())
     .then((response) => {
       form.querySelector('img.loading-img').classList.toggle('hidden');
-      dataContainer.innerHTML = buildHTML(response, typeOf);
+
+      let revealed;
       const topLevelData = document.getElementById('response-templates-data');
-      topLevelData.innerHTML = dataContainer.parentElement.innerHTML;
+
+      for (const typesOfContainers of typesOf) {
+        const [typeOf, containedIn] = typesOfContainers;
+
+        if (typeof response[typeOf] != 'undefined') {
+          const dataContainer = form.getElementsByClassName(containedIn)[0];
+          dataContainer.innerHTML = buildHTML(response[typeOf], typeOf);
+
+          if (!revealed) {
```

Comment:
Personal preference, non-blocking.  When I see `if !A { f(x) } else { g(x) }` my preference is to remove the negation and swap the conditionals (e.g. `if A { g(x) } else { f(x) }`)